feat: user input documentation (#120)
UnumID/UnumID.github.io: docs/api-overview.mdx

}
```

> - `PhoneCredential` is a "[single attribute atomic credential](/schema#schema-definitions)" and therefore does not have any sub-credentials. Because we specified a `allowUserInput` attribute, user will be allowed to prompt its own information for the given credential."

Reviewer: ```suggestion
- `PhoneCredential` is a "[single attribute atomic credential](/schema#schema-definitions)" and therefore does not have any sub-credentials. Because we set a `allowUserInput` attribute of `true`, the user will be given an opportunity to enter a new value if they choose."
```

```json title="PhoneCredential"

> {

Reviewer 2: The `mandatory` attribute is missing

Author: There is a task to make a combined check to allow user to input instead of just `allowUserInput`: https://trello.com/c/D4uhDaWt/5971-allow-user-to-input-only-if-allowuserinput-is-true-and-mandatory-is-ifavailable-changes-to-new-wallet-ui

Reviewer 2: @iagormoraes so you are saying `mandatory` is not needed if `allowUserInput` is `true`? If that is the case, could you please explain that in the docs? Either way, I would normally lean towards being more explicit when it comes to inputs 
like this. The implicit nature is where confusion happens. 

_Apologies this PR has sat so long open!_

    "type": "PhoneCredential",

>     "issuers": [],

Reviewer 2: ```suggestion
    "issuers": [],
    "mandatory": "if_available",
```

}
```

- `PhoneCredential` is a "[single attribute atomic credential](/schema#schema-definitions)" and therefore does not have any sub-credentials. Because we set a `allowUserInput` attribute of `true`, the user will be given an opportunity to enter a new value if they choose."

```json title="PhoneCredential"
{
    "type": "PhoneCredential",
    "issuers": [],
    "mandatory": "if_available",
    "allowUserInput": true
}
```

> Check that the `mandatory` is not present, the reason is when `allowUserInput`, the credential may be created and it is optional.

Reviewer 2: ```suggestion
Note that the `mandatory` is not present. This is thanks to `allowUserInput` being `true`, marking that the credential may be created and it is optional.
```

Reviewer 2: Although I really do not like this implicit behavior. I would really prefer the `mandatory` field needs to be a particular value when `allowUserInput` is used.

Reviewer: Why would `mandatory` need to be a particular value when `allowUserInput` is used? shouldn't the two be completely independent?

Author: I'm going to remove this part since `allowUserInput` is unrelated to mandatory.
